packages/ui-billboard/src/Billboard/props.ts

```diff
@@ -37,93 +37,94 @@ import type {
   OtherHTMLAttributes,
   PropValidators
 } from '@instructure/shared-types'
-
+import type { ViewProps } from '@instructure/ui-view'
+import { MouseEvent } from 'react'
 type BillboardOwnProps = {
-  hero?: React.ReactElement | ((...args: any[]) => any)
-  size?: 'small' | 'medium' | 'large'
-  as?: AsElementType
-  elementRef?: (element: Element | null) => void
-  heading?: string
-  headingAs?: 'h1' | 'h2' | 'h3' | 'span'
-  headingLevel?: 'h1' | 'h2' | 'h3' | 'h4'
-  message?: React.ReactNode | ((...args: any[]) => any)
-  onClick?: (...args: any[]) => any
-  href?: string
-  disabled?: boolean
-  readOnly?: boolean
-  margin?: Spacing
-}
-
-type PropKeys = keyof BillboardOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type BillboardProps = BillboardOwnProps &
-  WithStyleProps<BillboardTheme, BillboardStyle> &
-  OtherHTMLAttributes<BillboardOwnProps>
-
-type BillboardStyle = ComponentStyle<
-  'billboard' | 'content' | 'hero' | 'heading' | 'message'
->
-
-const propTypes: PropValidators<PropKeys> = {
   /**
    * Provide an <Img> component or Instructure Icon for the hero image
    */
-  hero: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+  hero?: React.ReactElement | ((size: 'medium' | 'x-large' | 'large') => string)
   /**
    * If you're using an icon, this prop will size it. Also sets the font-size
    * of the headline and message.
    */
-  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  size?: 'small' | 'medium' | 'large'
   /**
    * the element type to render as
    */
-  as: PropTypes.elementType,
+  as?: AsElementType
   /**
    * provides a reference to the underlying html root element
    */
-  elementRef: PropTypes.func,
+  elementRef?: (element: Element | null) => void
   /**
    * The headline for the Billboard. Is styled as an h1 element by default
    */
-  heading: PropTypes.string,
+  heading?: string
   /**
    * Choose the appropriately semantic tag for the heading
    */
-  headingAs: PropTypes.oneOf(['h1', 'h2', 'h3', 'span']),
+  headingAs?: 'h1' | 'h2' | 'h3' | 'span'
   /**
    * Choose the font-size for the heading (see the Heading component)
    */
-  headingLevel: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']),
+  headingLevel?: 'h1' | 'h2' | 'h3' | 'h4'
   /**
    * Instructions or information for the Billboard. Note: you should not pass
    * interactive content to this prop if you are also providing an `href` or
    * `onClick`. That would cause the Billboard to render as a button or link
    * and would result in nested interactive content.
    */
-  message: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+  message?: React.ReactNode
```

Comment:
It allows a function as well, with no parameters. The `callRenderProp` will handle it in the `index.tsx` :)

```suggestion
  message?: React.ReactNode | (() => React.ReactNode)
```